### ISSUE TITLE
Public API?

### ISSUE THREAD

**Issue by briantopping** [mikefarah/yq#831]
**Please describe your feature request.**
Hi I love `yq`, appreciate that the syntax is well-known. Would like to integrate it into a golang app I am writing.

I'm sure it's simple enough to grab some API and link it in, but there's no stability guarantee there. 

Any chance that a public API could be created and documented?

**Comment by mikefarah** [comment]
it's on my list of things to do :)

shouldn't be too far off having something - though it would be good to have input. I'm thinking of basically exposing the stream_evaluator and the all_at_once_evaluator, which basically lets you run expressions against yaml (via filenames, or directly as nodes).

Is that what you are looking for?

**Comment by briantopping** [comment]
Yes, I am looking to provide the capability to dynamically search a YAML tree. [This code](https://github.com/briantopping/cloud-provider-kubevirt/blob/cde831f54666c5348e5cce22bd3bc2fb6b632167/pkg/cloudprovider/kubevirt/loadbalancer.go#L328-L361) is an awkward implementation of what I suspect could be done very elegantly with `yq`.

**Comment by tamj0rd2** [comment]
I found this which shows an example of how to query a basic expression https://gist.github.com/mkatychev/80138136f05fd9cbbcb9b88047cba7ca
